src/CSModule.sol

```diff
@@ -713,10 +714,28 @@ contract CSModule is
         });
     }
 
+    /// @inheritdoc ICSModule
+    function onValidatorSlashed(
+        uint256 nodeOperatorId,
+        uint256 keyIndex
+    ) external onlyRole(VERIFIER_ROLE) {
+        _onlyExistingNodeOperator(nodeOperatorId);
+        NodeOperator storage no = _nodeOperators[nodeOperatorId];
+        if (keyIndex >= no.totalDepositedKeys) {
+            revert SigningKeysInvalidOffset();
+        }
+
+        uint256 pointer = _keyPointer(nodeOperatorId, keyIndex);
+        _isValidatorSlashed[pointer] = true;
```

Comment:
Should revert if it is already reported as slashed before.

Reply:
Why allow this and emit the event every time? It's misleading.

Reply:
Why is it misleading? The event is correct; it has no other side effects and can’t be used for griefing. We’re tight on bytecode, so I’d omit it for now.